pyanaconda/core/configuration.py

```diff
@@ -0,0 +1,133 @@
+               "{}".format(self._filename, super().__str__())
+
+
+class ConfigurationDataError(ConfigurationError):
+    """An error in the configuration data."""
+
+    def __init__(self, msg, section, option):
+        super().__init__(msg)
+        self._section = section
+        self._option = option
+
+    def __str__(self):
+        return "The following error has occurred while handling the option '{}' in the section " \
+               "'{}': {}".format(self._option, self._section, super().__str__())
+
+
+def create_parser():
+    """Create a new config parser.
+
+    :return: an instance of ConfigParser
+    """
+    return configparser.ConfigParser()
+
+
+def read_config(parser, path):
+    """Read a configuration file.
+
+    :param parser: an instance of ConfigParser
+    :param path: a path to the file
+    :raises: ConfigurationFileError
+    """
+    try:
+        with open(path, "r") as f:
+            parser.read_file(f, path)
+
+    except (configparser.Error, IOError) as e:
+        raise ConfigurationFileError(str(e), path)
+
+
+def write_config(parser, path):
+    """Write a configuration file.
+
+    :param parser: an instance of ConfigParser
+    :param path: a path to the file
+    :raises: ConfigurationFileError
+    """
+    try:
+        with open(path, "w") as f:
+            parser.write(f)
+
+    except (configparser.Error, IOError) as e:
+        raise ConfigurationFileError(str(e), path)
+
+
+def get_option(parser, section_name, option_name, converter=None):
+    """Get a converted value of the option.
+
+    :param parser: an instance of ConfigParser
+    :param section_name: a section name
+    :param option_name: an option name
+    :param converter: a function or None
+    :return: a converted value
+    :raises: ConfigurationAccessError
+    """
+    try:
+        # Return a string.
+        if converter is None:
+            return parser.get(section_name, option_name)
+
+        # Return a boolean.
+        if converter is bool:
+            return parser.getboolean(section_name, option_name)
+
+        # Return a converted value.
+        return converter(parser.get(section_name, option_name))
+
+    except (configparser.Error, ValueError) as e:
+        raise ConfigurationDataError(str(e), section_name, option_name)
+
+
+def set_option(parser, section_name, option_name, value):
+    """Set the option.
+
+    :param parser: an instance of ConfigParser
+    :param section_name: a section name
+    :param option_name: an option name
+    :param value: an option value
+    :raises: ConfigurationAccessError
+    """
+    try:
+        # Make sure that the section and option exists.
+        parser.get(section_name, option_name)
+
+        # Set the option to the string representation of the value.
+        parser[section_name][option_name] = str(value)
+
+    except (configparser.Error, ValueError) as e:
+        raise ConfigurationDataError(str(e), section_name, option_name)
```

Comment:
Sorry, but I'm not able to see why is there API for `pyanaconda/configuration.py` and here. What is the benefit why we have two APIs? Wouldn't it be better to convert `pyanaconda/configuration.py` to a module (python module) and have this as an internal implementation details, so to provide only the `pyanaconda/configuration.py` API?